pkg/action/codefresh.go

```diff
@@ -320,6 +321,66 @@ func (o *CfApply) ApplyCodefresh() error {
 	return nil
 }
 
+func (o *CfApply) ApplyBackupMgr() error {
+	valsX := objx.New(o.vals)
+
+	mongoURI := valsX.Get(c.BkpManagerMongoURI).Str()
+	if mongoURI == "" {
+		info("Mongo URI is not specified, trying to get it automatically from the installed Codefresh release")
+		var err error
+		mongoURI, err = o.getMongoURIFromRelease()
+		if err != nil {
+			return err
+		}
+	}
+
+	debug("Mongo URI is: %s", mongoURI)
+	valsX.Set(c.BkpManagerMongoURI, mongoURI)
+
+	installerType := valsX.Get(c.KeyInstallerType).String()
+	if installerType == installerTypeHelm {
+		helmChartName := valsX.Get(c.KeyHelmChart).String()
+		helmReleaseName := valsX.Get(c.KeyHelmRelease).Str(kindBackupManager)
+		rel, err := DeployHelmRelease(
+			helmReleaseName,
+			helmChartName,
+			valsX,
+			o.cfg,
+			o.Helm,
+		)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to deploy %s chart", helmChartName)
+		}
+		PrintHelmReleaseInfo(rel, c.Debug)
+		info("\n%s has been deployed to namespace %s\n", helmReleaseName, o.Helm.Namespace)
+		return nil
+	}
+	return fmt.Errorf("Wrong installer type %s", installerType)
+}
+
+func (o *CfApply) getMongoURIFromRelease() (string, error) {
+
+	cfRelVals, err := GetReleaseValues(codefreshHelmReleaseName, o.cfg)
+	if err != nil {
+		return "", err
+	}
+
+	cfRelValsX := objx.New(cfRelVals)
+
+	runtimeMongoURI := cfRelValsX.Get(c.ReleaseMongoURI).Str()
+	mongoRootUser := cfRelValsX.Get(c.ReleaseRootUser).Str()
+	mongoRootPassword := cfRelValsX.Get(c.ReleaseRootPassword).Str()
+
+	debug("runtimeMongoURI: %s, mongoRootUser: %s, mongoRootPassword: %s", runtimeMongoURI, mongoRootUser, mongoRootPassword)
+
+	if runtimeMongoURI == "" || mongoRootUser == "" || mongoRootPassword == "" {
+		return "", fmt.Errorf("Failed to get the mongo URI value")
+	}
+
+	rootMongoURI := "mongodb://" + mongoRootUser + ":" + mongoRootPassword + "@" + strings.Split(runtimeMongoURI, "@")[1]
```

Comment:
If MongoURI does not include "@" it will be panic. please add some check and return error if cannot parse uri
also please remove printing db password even in debug

Reply:
@kosta709 I have just fixed it